Sender/BMS_Sender.c

```diff
@@ -0,0 +1,88 @@
+/*--------------------------------------------------------------------------------
+---    \file       BMS_Sender.c
+---   \author      Naveen Balachandar (RBEI/ECK3) 
+---
+---     \brief     proj file to stream BMS data
+------------------------------------------------------------------------------------*/
+/*------ standard includes -------*/
+#include <stdio.h>
+#include <stdlib.h>
+
+/*------ Project includes -------*/
+#include "BMS_Sender.h"
+
+/*Intial states of operation*/
+streamAlert_Status_s streamAlertStat = {NOT_SENT,FILE_ACCESS_FAILURE};
+
+/*---------------------------------------------------------------------------*/
+/*     FUNCTION:    streamFileInpData
+ */
+/*!    \brief        read input data from file and stream continuosly
+ * 
+ *     \param       None
+ *     \returns     streamAlert_Status_s -success/fail
+ *
+*//*------------------------------------------------------------------------*/
+streamAlert_Status_s  streamFileInpData(char *inpFilePath)
+{
+  FILE *fptr;
+  unsigned int dataIndex_ui;
+  float tempRead ,chargerateRead;
+  /*Read the input file*/
+  fptr =fopen(inpFilePath,"r");
+
+  if ( NULL != fptr) /*validating file access*/
+  { 
+     /*msg that file access is success*/
+     streamAlertStat.FileReadStatus = FILE_ACCESS_SUCCESS; 
+     printf("Data Read from file as follows\n");
+     for(dataIndex_ui=0;fscanf(fptr,"%f\t%f", &tempRead,&chargerateRead) != EOF;dataIndex_ui++)
```

Comment:
What if one float was read but not another? That condition will be hard to handle and test here. That would be a reason to simplify the code.